src/MICore/UnixUtilities.cs

```diff
@@ -160,5 +163,28 @@ internal static bool IsProcessRunning(int processId)
             // if there is no process with the ID specified.
             return UnixNativeMethods.GetPGid(processId) >= 0;
         }
+
+        public static bool IsBinarySigned(string filePath)
```

Comment:
I was thinking of using a static. But thinking about it more, since VS Code will recreate the process each time, that isn't going to help.

Do we know what all codesign will do to check the signature? Codesign checking on Windows is quite an expensive operation as it needs to read every byte of the file and hash it. Then it needs to go off to web services to check if the certificate has been revoked. Do we know if this will do anything like that?